### ISSUE TITLE
Cannot set property 'defaultProviders' of undefined

### ISSUE THREAD

**Issue by harshadyeola** [aws/aws-sdk-js#1306]
Facing following issue from `aws-sdk`  with latest version `2.7.22`
```
 server.js:72 () TypeError: Cannot set property 'defaultProviders' of undefined
    at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/lib/node_loader.js:33:46)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/clients/sts.js:1:63)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/lib/credentials/temporary_credentials.js:2:11)
    at Module._compile (module.js:456:26)
```

**Comment by chrisradek** [comment]
@harshadyeola 
When does this issue occur? Does it happen just when requiring the SDK? If not, can you share a code snippet that reproduces the issue?

**Comment by harshadyeola** [comment]
@chrisradek sure. here is code snippet where it fails 
```
function fetchScheduledPayload(key) {
    logger.debug('inside fetchScheduledPayload, key:', key);
    var s3 = new AWS.S3();
    var getObjectAsync= Promise.promisify(s3.getObject, s3);
    return getObjectAsync({
        Bucket: process.env.StoreBucket,
        Key: key+'.json'
    });
}
```

**Comment by chrisradek** [comment]
Which `Promise` library are you using? Can you also share how you're setting credentials?

You might also want to try calling the `promise()` method on the operation directly, instead of Promisifying it with another library:
```javascript
function fetchScheduledPayload(key) {
    logger.debug('inside fetchScheduledPayload, key:', key);
    var s3 = new AWS.S3();
    return s3.getObject({
        Bucket: process.env.StoreBucket,
        Key: key+'.json'
    }).promise();
}
```

**Comment by harshadyeola** [comment]
@chrisradek above solution did not work.

**Comment by harshadyeola** [comment]
@chrisradek we are using `bluebird` library for Promise.

**Comment by chrisradek** [comment]
@harshadyeola 
Can you share how you're setting up your credentials (leaving out any secrets of course).

**Comment by harshadyeola** [comment]
@chrisradek Actually bucket access is available to instance via Instance Role. so no other authentication methods are setup for the application.

**Comment by ysg410** [comment]
@chrisradek

**Comment by ysg410** [comment]
@chrisradek 
We, @harshadyeola and I also encountered the following issue along with ```defaultProviders``` issue. Attaching the error messages : 
```
server.js:221 () TypeError: Cannot read property 'prototype' of undefined
	at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/lib/node_loader.js:46:27)
	at Module._compile (module.js:456:26)
	at Object.Module._extensions..js (module.js:474:10)
	at Module.load (module.js:356:32)
	at Function.Module._load (module.js:312:12)
	at Module.require (module.js:364:17)
	at require (module.js:380:17)
	at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/clients/sts.js:1:63)
	at Module._compile (module.js:456:26)
	at Object.Module._extensions..js (module.js:474:10)
	at Module.load (module.js:356:32)
	at Function.Module._load (module.js:312:12)
	at Module.require (module.js:364:17)
	at require (module.js:380:17)
	at Object.<anonymous> (/apps/workers/node_modules/aws-sdk/lib/credentials/temporary_credentials.js:2:11)
	at Module._compile (module.js:456:26)
```

**Comment by harshadyeola** [comment]
@chrisradek seems like this is fixed in `2.7.24`. So closing this issue.
